packages/calcite-components/src/components/tab-title/tab-title.scss

```diff
@@ -259,101 +183,198 @@
-  }
-
-  & calcite-icon {
-    color: inherit;
-  }
-}
-
-// compensate for spacing when no hastext and two icons
-:host([icon-start][icon-end]) {
-  .calcite-tab-title--icon:first-child {
-    margin-inline-end: var(--calcite-spacing-sm);
+    color: var(--calcite-tab-close-icon-color-press, var(--calcite-color-text-1));
+    background-color: var(--calcite-tab-close-icon-background-color-press, var(--calcite-color-foreground-3));
   }
 }
 
 // bordered styles
 :host([bordered]) .container {
+  border-color: transparent;
+  border-inline-width: var(--calcite-spacing-px);
+  border-inline-style: solid;
+
   &:hover,
   &:active {
     @apply text-color-1;
   }
 
-  &:not(.container--bottom) .close-button {
-    block-size: calc(100% - var(--calcite-spacing-px));
-    margin-block-start: -1px;
-  }
-  .close-button {
+  .x-button {
     & calcite-icon {
       margin-block-start: var(--calcite-spacing-px);
     }
 
     &:focus,
     &:hover,
     &:active {
-      box-shadow: 0 2px 0 0 var(--calcite-color-foreground-3);
+      box-shadow: 0 2px 0 0 var(--calcite-tab-close-icon-background-color-press, var(--calcite-color-foreground-3));
     }
   }
 
-  &.container--bottom .close-button {
-    box-shadow: 0 -2px 0 0 transparent;
-    & calcite-icon {
-      margin-block-end: var(--calcite-spacing-px);
+  &.container--bottom {
+    border-block-start-style: solid;
+    border-block-start-width: 1px;
+
+    .selected-indicator {
+      inset-block-start: unset;
+      inset-block-end: 0;
     }
-    &:focus,
-    &:hover,
-    &:active {
-      box-shadow: 0 -2px 0 0 var(--calcite-color-foreground-3);
+
+    .x-button {
+      box-shadow: 0 -2px 0 0 var(--calcite-tab-close-icon-background-color, var(--calcite-color-transparent));
+
+      &:focus,
+      &:hover,
+      &:active {
+        box-shadow: 0 -2px 0 0 var(--calcite-tab-close-icon-background-color-press, var(--calcite-color-foreground-3));
+      }
+
+      calcite-icon {
+        margin-block-end: var(--calcite-spacing-px);
+      }
+    }
+  }
+
+  &:not(.container--bottom) {
+    border-block-end-style: solid;
+    border-block-end-width: 1px;
+
+    .x-button {
+      block-size: calc(100% - var(--calcite-spacing-px));
+      margin-block-start: -1px;
+    }
+  }
+
+  .selected-indicator {
+    inset-block-start: 0;
+    inset-block-end: unset;
+    inset-inline-start: -1px;
+    inset-inline-end: 0;
+    inline-size: calc(100% + var(--calcite-spacing-base));
+  }
+
+  &:host([selected]) {
+    &::after {
+      @apply absolute
+          block
+          w-full
+          h-0.5
+          transition-default;
+      inset-block-end: -1px;
+      inset-inline-start: 0;
+      inset-inline-end: 0;
+      inline-size: 100%;
+      content: "";
+    }
+
+    &.container--bottom::after {
+      inset-block-start: -1px;
+    }
+
+    &:host(:hover) .container::after {
+      background-color: var(--calcite-tab-background-color, var(--calcite-color-foreground-2));
```

Comment:
I think this should be  `--calcite-tab-background-color-hover` to customize the hover background-color which has a different default value. 

Applies to other places in this file.